File: Data Structures & Algorithms/string-encode-and-decode/submission-0.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        enc_str = ""

        for i in strs:
            enc_str+= f"{len(i)}:{i}"
        return enc_str

        return f"{strs[0]}{strs[1]}"
    def decode(self, s: str) -> List[str]:
        retval = []
        while s != "":

            length_str = ""
            c = s[0]

            while c != ":":
                length_str += c
                s=s[1:]
                c = s[0]

            

            length = int(length_str)
            s=s[1:]
            retval.append(s[:length])
            s = s[length:]
        return retval

File: Data Structures & Algorithms/anagram-groups/submission-0.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string, vector<string>> anagram_matches;

        for (string word : strs){
            string srt_wrd = word;
            sort(srt_wrd.begin(),srt_wrd.end());
            anagram_matches[srt_wrd].push_back(word);
        }

        vector<vector<string>> retval;

        for (auto l : anagram_matches){
            retval.push_back(l.second);
        }

        return retval;
    }
};
